### ISSUE TITLE
Can't use --memory-map and --debug-mode options simultaneously

### ISSUE THREAD

**Issue by huwan** [facebookresearch/dlrm#50]
The following command crashes when using both `--memory-map` and `--debug-mode` options. (The command works if I disable one of them.)

> python ../dlrm/dlrm_s_pytorch.py --arch-sparse-feature-size=16 --arch-mlp-bot='13-512-256-64-16' --arch-mlp-top='512-256-1' --data-generation=dataset --data-set=kaggle --raw-data-file=./data.dlrm/train.txt --processed-data-file=./data.dlrm/kaggleAdDisplayChallenge_processed.npz --loss-function=bce --round-targets=True --learning-rate=0.1 --mini-batch-size=128 --print-freq=1024 --print-time --test-mini-batch-size=16384 --test-num-workers=8 --test-freq=1024 --load-model=./data.dlrm/dlrm.model.pt --inference-only `--memory-map` `--debug-mode`

Output:
```
Using CPU...
Reading pre-processed data=./data/kaggleAdDisplayChallenge_processed.npz
Sparse fea = 26, Dense fea = 13
Defined train indices...
Randomized indices across days ...
......
Loading saved model ./data.dlrm/dlrm.model.pt

Saved at: epoch = 0/1, batch = 299000/306969, ntbatch = 200
Training state: loss = 0.450191, accuracy = 79.028 %
Testing state: loss = 0.452747, accuracy = 78.865 %
time/loss/accuracy (if enabled):
Traceback (most recent call last):
  File "../dlrm/dlrm_s_pytorch.py", line 850, in <module>
    for j, (X, lS_o, lS_i, T) in enumerate(train_ld):
  File "/home/test/anaconda3/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 560, in __next__
    batch = self.collate_fn([self.dataset[i] for i in indices])
  File "/home/test/anaconda3/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 560, in <listcomp>
    batch = self.collate_fn([self.dataset[i] for i in indices])
  File "/home/test/dlrm/dlrm_data_pytorch.py", line 292, in __getitem__
    return self.X_int[i], self.X_cat[i], self.y[i]
IndexError: index -32743299 is out of bounds for axis 0 with size 6548659
```

Thanks.

**Comment by mnaumovfb** [comment]
Thank you for reporting this issue. I have fixed it with the latest commit 7c55962. 

The reason for this bug was that --debug-mode flag triggered an additional pass over the data, which was not handled correctly with --memory-map flag in the default data loader. 

Please give it a try and let me know if it works for you.

**Comment by huwan** [comment]
It works, thanks. Closing.
